### ISSUE TITLE
OCMock revision different in dependency "GoogleUtilities"

### ISSUE THREAD

**Issue by guidedways** [firebase/firebase-ios-sdk#9591]
`firebase-ios-sdk` `Package.swift` contains a dependency on OCMock with a different revision number than that in `GoogleUtilities` (another dependency). When you try and resolve package dependencies, xcode at times complains that the two revisions do not match and fails to resolve dependencies.

firebase-ios-sdk uses:

```
    .package(
      name: "OCMock",
      url: "https://github.com/erikdoe/ocmock.git",
      .revision("c5eeaa6dde7c308a5ce48ae4d4530462dd3a1110")
    ),
```

while GoogleUtilities uses:

```
.package(
      name: "OCMock",
      url: "https://github.com/firebase/ocmock.git",
      .revision("7291762d3551c5c7e31c49cce40a0e391a52e889")
    ),
```

**Comment by paulb777** [comment]
@guidedways Thanks for the report. It's interesting that we haven't seen a problem with the two different references.  

Would you explain more about your use case? Is there a consistent way to reproduce?

Either way, we can unify these in an upcoming release.

**Comment by paulb777** [comment]
It looks like when we updated the Firebase repo for https://github.com/erikdoe/ocmock/pull/499, we did not update GoogleUtilities similarly

**Comment by guidedways** [comment]
As you're aware, Swift PM support in Xcode is still _meh_. We only came across this recently when we tried to speed up package resolution. Instead of letting Xcode manage package resolution for us (and have it frequently refresh and fail for no obvious reason at times), we did the following:

1) Created a local `Dependencies` folder 
2) Manually cloned all our dependencies (i.e. `git clone https://path.to.package`)
3) Dragged these into Xcode's workspace to create local file references

Using local file references for each package we were able to speed up package resolution. This way deleting `DerivedData` (if needed), performing a clean build or simply relaunching Xcode would no longer resolve each of these packages - it would only resolve _sub-dependencies_ declared within these packages.

We noticed we could speed this up further by cloning sub-dependencies and including those as file references too. This worked for all our package dependencies except for FireBase - when we cloned `GoogleUtilities`, the following happened:

`> xcodebuild -resolvePackageDependencies -workspace ... -scheme "..."`

Output:

```
ocmock is required using two different revision-based requirements (c5eeaa6dde7c308a5ce48ae4d4530462dd3a1110 and 7291762d3551c5c7e31c49cce40a0e391a52e889), which is not supported
```

And that's how we found out that firebase-ios-sdk had a different revision to GoogleUtilities. `xcodebuild` wasn't happy about this.
